lib/python/bind_operators.h

```diff
@@ -83,6 +83,90 @@ void bind_astype(py::class_<T, Ignored...> &c) {
         :rtype: Union[scipp.Variable, scipp.DataArray])");
 }
 
+template <class T, class... Ignored>
+void bind_to(py::class_<T, Ignored...> &c) {
+  c.def(
+      "to",
+      [](const T &self, const py::object &dt,
+         const std::optional<ProtoUnit> unit, const bool copy) {
+        using scipp::core::dtype;
+
+        const auto src_dtype(self.dtype());
+        const auto dest_dtype(dt.is_none() ? src_dtype : scipp_dtype(dt));
+        const auto dest_units(unit.has_value() ? make_unit(unit.value())
+                                               : self.unit());
+
+        [[maybe_unused]] py::gil_scoped_release release;
+
+        bool convert_dtype_first;
+
+        if (src_dtype == dest_dtype) {
+          return to_unit(self, dest_units,
+                         copy ? scipp::CopyPolicy::Always
+                              : scipp::CopyPolicy::TryAvoid);
+        } else if (dest_dtype == dtype<double>) {
+          convert_dtype_first = true;
+        } else if (src_dtype == dtype<double>) {
+          convert_dtype_first = false;
+        } else if (dest_dtype == dtype<float>) {
+          convert_dtype_first = true;
+        } else if (src_dtype == dtype<float>) {
+          convert_dtype_first = false;
+        } else if (src_dtype == dtype<int64_t> &&
+                   dest_dtype == dtype<int32_t>) {
+          convert_dtype_first = false;
+        } else if (src_dtype == dtype<int32_t> &&
+                   dest_dtype == dtype<int64_t>) {
+          convert_dtype_first = true;
+        } else {
+          convert_dtype_first = true;
+        }
+
+        if (convert_dtype_first) {
+          auto data = astype(self, dest_dtype,
+                             copy ? scipp::CopyPolicy::Always
+                                  : scipp::CopyPolicy::TryAvoid);
```

Comment:
I think I would recommend moving to implement of `to` to Python, so we can avoid a number of potentially error prone repetitions, such as setting the copy policy from the bool `copy` param.